packages/fax/src/rest/v3/faxes/faxes-api.ts

```diff
@@ -0,0 +1,273 @@
+    super(sinchClientParameters, 'FaxesApi');
+  }
+
+  /**
+   * Delete fax content
+   * Delete the fax content for a fax using the ID number of the fax. Please note that this only deletes the content of the fax from storage.
+   * @param { DeleteFaxContentRequestData } data - The data to provide to the API call.
+   */
+  public async deleteContent(data: DeleteFaxContentRequestData): Promise<void> {
+    this.client = this.getSinchClient();
+    const getParams = this.client.extractQueryParams<DeleteFaxContentRequestData>(data, [] as never[]);
+    const headers: { [key: string]: string | undefined } = {
+      'Content-Type': 'application/json',
+      'Accept': 'application/json',
+    };
+
+    const body: RequestBody = '';
+    const basePathUrl = `${this.client.apiClientOptions.basePath}/v3/projects/${this.client.apiClientOptions.projectId}/faxes/${data['id']}/file`;
+
+    const requestOptions
+      = await this.client.prepareOptions(basePathUrl, 'DELETE', getParams, headers, body || undefined);
+    const url = this.client.prepareUrl(requestOptions.basePath, requestOptions.queryParams);
+
+    return this.client.processCall<void>({
+      url,
+      requestOptions,
+      apiName: this.apiName,
+      operationId: 'DeleteFaxContentById',
+    });
+  }
+
+  /**
+   * Download fax content
+   * Download the fax content.
+   * @param { DownloadFaxContentRequestData } data - The data to provide to the API call.
+   */
+  public async downloadContent(data: DownloadFaxContentRequestData): Promise<FileBuffer> {
+    this.client = this.getSinchClient();
+    data['fileFormat'] = data['fileFormat'] !== undefined ? data['fileFormat'] : 'pdf';
+    const getParams = this.client.extractQueryParams<DownloadFaxContentRequestData>(data, [] as never[]);
+    const headers: { [key: string]: string | undefined } = {
+      'Content-Type': 'application/json',
+      'Accept': 'application/pdf',
+    };
+
+    const body: RequestBody = '';
+    const basePathUrl = `${this.client.apiClientOptions.basePath}/v3/projects/${this.client.apiClientOptions.projectId}/faxes/${data['id']}/file.${data['fileFormat']}`;
+
+    const requestOptions = await this.client.prepareOptions(basePathUrl, 'GET', getParams, headers, body || undefined);
+    const url = this.client.prepareUrl(requestOptions.basePath, requestOptions.queryParams);
+
+    return this.client.processFileCall({
+      url,
+      requestOptions,
+      apiName: this.apiName,
+      operationId: 'GetFaxFileById',
+    });
+  }
+
+  /**
+   * Get fax
+   * Get fax information using the ID number of the fax.
+   * @param { GetFaxRequestData } data - The data to provide to the API call.
+   */
+  public async get(data: GetFaxRequestData): Promise<Fax> {
+    this.client = this.getSinchClient();
+    const getParams = this.client.extractQueryParams<GetFaxRequestData>(data, [] as never[]);
+    const headers: { [key: string]: string | undefined } = {
```

Comment:
Correct me if I'm wrong but it looks like all methods within this class have the same `headers` var defined?
Maybe a class variable would be more concise?

Reply:
This is because the specification defines only endpoint with these headers. This code is generated from the specification and it's not a problem to have these variables repeated. Conciseness is not sought here: if the specification changes, it will be over-complicated in the mustache templates to revert the assignment of a variable

Reply:
Code duplication is simply eye catching :)  It's up to you to decide then.